src/main/java/org/generationcp/middleware/operation/builder/ExperimentBuilder.java

```diff
@@ -223,9 +224,21 @@ private Experiment createExperiment(final ExperimentModel experimentModel, final
 		experiment.setVariates(this.getVariates(experimentModel, variableTypes));
 		experiment.setLocationId(experimentModel.getGeoLocation().getLocationId());
 		experiment.setObsUnitId(experimentModel.getObsUnitId());
+		experiment.setNameValueMap(this.getNameByGid(experimentModel));
 		return experiment;
 	}
 
+	private Map<Integer, String> getNameByGid(final ExperimentModel experimentModel) {
```

Comment:
rename to something more relevant where we can understand what the map is refering to